### PR TITLE
Modify len with shape and index.

### PR DESCRIPTION
**Pull request by pashu123**
This makes it easy for the downstream compilers like MLIR to trace the code.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by pashu123** [comment]
@LukeWood @fchollet PTAL.

**Comment by LukeWood** [comment]
Please review the CLA!

Also, can you provide some more info on this?

**Comment by pashu123** [comment]
> Please review the CLA!
> 
> Also, can you provide some more info on this?

Thanks, I have reviewed the CLA. So, we were trying to use the runtime  https://github.com/iree-org/iree which in turn calls https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/mlir for tracing the tensorflow computation graph. The problem is `len()` is not supported under tracing and hence it can be replaced by `.shape[0]`.

**Comment by pashu123** [comment]
@LukeWood PTAL.

**Comment by LukeWood** [comment]
does this not change the behavior?  Is the original implementation bugged?  

```
broadcast_shape = [1] * input_shape.shape[0]
```

is NOT the same as len(input_shape)!

**Comment by pashu123** [comment]
> does this not change the behavior? Is the original implementation bugged?
> 
> ```
> broadcast_shape = [1] * input_shape.shape[0]
> ```
> 
> is NOT the same as len(input_shape)!

Do you mean `input_shape.shape[0]` is not same as `len(input_shape)`? Well if input_shape is a 1d tensor (input_shape is indeed a 1d tensor) they are the same?

**Comment by LukeWood** [comment]
> > does this not change the behavior? Is the original implementation bugged?
> > ```
> > broadcast_shape = [1] * input_shape.shape[0]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > is NOT the same as len(input_shape)!
> 
> Do you mean `input_shape.shape[0]` is not same as `len(input_shape)`? Well if input_shape is a 1d tensor (input_shape is indeed a 1d tensor) they are the same?

True, is our GroupNormalization only called on 1D tensors?  If so, then this PR is fine.

how about we have an if statement validating that the input is a 1D tensor.  This way, we can raise a value error if it’s not and know we’re using the layer incorrectly.  If you add a guard clause like that I can then pull the PR, run the tests, and merge this if it all passes.

**Comment by tanzhenyu** [comment]
Looks like there's merge conflict

**Comment by LukeWood** [comment]
> @LukeWood Why we have still a custom class here? Is this different from the Group normalization we have in Keras now? https://github.com/keras-team/keras/blob/master/keras/layers/normalization/group_normalization.py

Verisoning

**Comment by tanzhenyu** [comment]
This is fixed per https://github.com/keras-team/keras-cv/pull/1035
